R/get_draws.R

```diff
@@ -0,0 +1,399 @@
+#' @title Postprocess to generate a draws dataframe
+#'
+#' @description
+#' This function takes in the two input data sources, the CmdStan fit object,
+#' and the 3 relevant mappings from stan indices to the real data, in order
+#' to generate a dataframe containing the posterior draws of the counts (e.g.
+#' hospital admissions), the wastewater concentration values, the "global" R(t),
+#' and the "local" R(t) estimates + the critical metadata in the data.
+#' This funtion has a default method that takes the two sets of input data,
+#' the last of stan arguments, and the CmdStan fitting object, as well as an S3
+#' method for objects of class 'wwinference_fit'
+#'
+#'
+#' @param x Either a dataframe of wastewater observations, or an object of
+#' class wwinference_fit
+#' @param ... additional arguments
+#' @param what Character vector. Specifies the variables to extract from the
+#' draws. It could be any from `"all"` `"predicted_counts"`, `"predicted_ww"`,
+#' `"global_rt"`, or `"site_level_rt"`. When `what = "all"` (the default),
+#' the function will extract all four variables.
+#' @return  A tibble containing the full set of posterior draws of the
+#' estimated, nowcasted, and forecasted: counts, site-level wastewater
+#' concentrations, "global"(e.g. state) R(t) estimate, and the  "local" (site +
+#' the one auxiliary subpopulation) R(t) estimates. In the instance where there
+#' are observations, the data will be joined to each draw of the predicted
+#' observation to facilitate plotting.
+#' @export
+get_draws <- function(x, ..., what = "all") {
+  UseMethod("get_draws")
+}
+
+#' @rdname get_draws
+#' @details
+#' The function `get_draws_df()` has been deprecated in favor of `get_draws()`.
+#'
+#' @export
+get_draws_df <- function(x, ...) {
+  .Deprecated("get_draws")
+}
+
+#' S3 method for extracting posterior draws alongside data for a
+#' wwinference_fit object
+#'
+#' This method overloads the generic `get_draws` function specifically
+#' for objects of type 'wwinference_fit'.
+#'
+#' @rdname get_draws
+#' @export
+get_draws.wwinference_fit <- function(x, ..., what = "all") {
+  get_draws.data.frame(
+    x = x$raw_input_data$input_ww_data,
+    count_data = x$raw_input_data$input_count_data,
+    stan_data_list = x$stan_data_list,
+    fit_obj = x$fit,
+    what = what
+  )
+}
+
+#' @export
+#' @rdname get_draws
+get_draws.default <- function(x, ..., what = "all") {
+  stop(
+    "No method defined for get_draws for object of class(es) ",
+    paste(class(x), collapse = ", "),
+    ". Use directly on a wwinference_fit object or a",
+    "dataframe of wastewater observations.",
+    call. = FALSE
+  )
+}
+
+get_draws_what_ok <- c(
```

Comment:
Is it standard practice to just define a vector like this outside of a function in one of these files? Will defer to you but I haven't seen this before!